### PR TITLE
Better wording for the barrier.

### PR DESCRIPTION
**Pull request by dns2utf8**
with #56 in mind

### CODE REVIEW

**Comment by robinst** [comment]
Ah, now I see. I don't know why you would write it like that instead of using a channel (which does not have the problem), but that's a different topic.

It's a bit better with the PR, but still not 100% clear. Can you add some introductory text to that section as well? Something like this maybe:

> A way to wait for all the jobs to finish is using a barrier and waiting on it in the workers and the starter thread. But note that if you put more jobs ...

**Comment by frewsxcv** [comment]
Doesn't a deadlock usually imply that it's locked indefinitely? In this case, the barrier isn't locked indefinitely, right? Even though I [initially approved this text](https://github.com/frewsxcv/rust-threadpool/pull/35), I do think the usage of "deadlock" is a little confusing.

File: lib.rs [hunk 1]
```diff
@@ -56,6 +56,8 @@
 //! let pool = ThreadPool::new(n_workers);
 //! let an_atomic = Arc::new(AtomicUsize::new(0));
 //!
+//! assert!(n_jobs <= n_workers, "too many jobs, will deadlock");
```
Comment (dns2utf8) [inline lib.rs]: As long as this condition is matched, the deadlock is prevented.

**Comment by robinst** [comment]
Looks good to me now.

**Comment by frewsxcv** [comment]
Thanks!

**Pull request merged**